src/framework/components/joint/component.js

```diff
@@ -0,0 +1,371 @@
+import { Mat4 } from '../../../math/mat4.js';
+import { Quat } from '../../../math/quat.js';
+import { Vec2 } from '../../../math/vec2.js';
+
+import { Component } from '../component.js';
+
+/**
+ * @private
+ * @component
+ * @class
+ * @name JointComponent
+ * @augments Component
+ * @classdesc The JointComponent adds a physics joint constraint linking two rigid bodies.
+ * @param {JointComponentSystem} system - The ComponentSystem that created this Component.
+ */
+class JointComponent extends Component {
+    constructor(system, entity) {
+        super(system, entity);
+
+        // #ifdef DEBUG
+        if (typeof Ammo === 'undefined') {
+            console.error('ERROR: Attempting to create a pc.JointComponent but Ammo.js is not loaded');
+        }
+        // #endif
+
+        this._constraint = null;
+
+        this._entityA = null;
+        this._entityB = null;
```

Comment:
I wonder if this should use  EntityReference so that it works in the Editor? @vkalpias would know more.

https://github.com/playcanvas/engine/blob/192a6e870d832cc202a44ad06abf86677df580ab/src/framework/components/render/component.js#L89-L90